src/miner.cpp

```diff
@@ -624,6 +624,115 @@ bool CreateCoinStake( CBlock &blocknew, CKey &key,
+    // staking, and if so, it will pop off the single output and replace it with the output split among the
+    // calculated number of outputs to meet efficiency requirements.
+    
+    int64_t nTotalStakeOutputValue = 0;
+    CScript scriptPubKey;
+    unsigned int nUTXOCount = 1;
+    int64_t nActualStakeOutputValue = 0;
+
+    // Pull out total value of output for stake, which also includes (interest or CBR) and research rewards at this point.
+    nTotalStakeOutputValue = blocknew.vtx[1].vout[1].nValue;
+    nUTXOCount = GetNumberOfStakeOutputs(nTotalStakeOutputValue);
+    
+    // Only go to the effort to split if nUTXOCount > 1, otherwise leave undisturbed.
+    if (nUTXOCount > 1)
+    {
+        scriptPubKey = blocknew.vtx[1].vout[1].scriptPubKey;
+        
+        // Remove the existing single stake output.
+        blocknew.vtx[1].vout.pop_back();
+    
+        nActualStakeOutputValue = nTotalStakeOutputValue / nUTXOCount;
+
+        if (fDebug2) LogPrintf("SplitCoinStakeOutput: nUTXOCount = %f", nUTXOCount);
+        if (fDebug2) LogPrintf("SplitCoinStakeOutput: nActualStakeOutputValue = %f", CoinToDouble(nActualStakeOutputValue));
+        
+        int64_t nSumStakeUTXOValue = 0;
+        for (unsigned int i = 1; i < nUTXOCount; i++)
+        {
+            blocknew.vtx[1].vout.push_back(CTxOut(nActualStakeOutputValue, scriptPubKey));
+            LogPrintf("SplitCoinStakeOutput: create stake UTXO %i value %f", i, CoinToDouble(nActualStakeOutputValue));
+            nSumStakeUTXOValue += nActualStakeOutputValue;
+        }
+        // For the last UTXO, subtract the running sum from the desired total, which does not include the last UTXO so far,
+        // to recover anything lost from rounding. This will be a very small difference from the others.
+        // The reason we go through this trouble is that integer division rounds down, and we don't even want
+        // to have the wallet lose 1 Halford.
+        blocknew.vtx[1].vout.push_back(CTxOut((nTotalStakeOutputValue - nSumStakeUTXOValue), scriptPubKey));
+        LogPrintf("SplitCoinStakeOutput: create stake UTXO %i value %f", nUTXOCount, CoinToDouble(nTotalStakeOutputValue - nSumStakeUTXOValue));
+    }
+}
+
+
+
+unsigned int GetNumberOfStakeOutputs(int64_t nValue)
+{
+    int64_t nMinUTXOPostSplitValue = 0;
+    double dEfficiency = 0.0;
+    int64_t nDesiredStakeUTXOValue = 0;
+    unsigned int nUTXOCount = 1;
+
+    // For the definition of the constant G, please see
+    // https://docs.google.com/document/d/1OyuTwdJx1Ax2YZ42WYkGn_UieN0uY13BTlA5G5IAN00/edit?usp=sharing
+    // Refer to page 5 for G. This link is a draft of an upcoming bluepaper section.
+    const double G = 9942.2056;
+
+    // Pull Minimum Post Stake UTXO Split Value from config or command line parameter.
+    // Default to 800 and do not allow it to be specified below 800 GRC.
+    nMinUTXOPostSplitValue = GetArg("-minimumstakeutxovalueforsplit", (int64_t)800) * COIN;
```

Comment:
There's a small bug here where the lower cap is 800 Halfords, not coins, if min stake value is set to 0. You could also extract the constant to avoid someone changing one but not the other.
```
namespace
{
    const uint64_t MIN_STAKE_SPLIT_VALUE_GRC = 800;
}

// ...
const int64_t = std::max(
        GetArg("-minimumstakeutxovalueforsplit", MIN_STAKE_SPLIT_VALUE_GRC),
        MIN_STAKE_SPLIT_VALUE_GRC) * COIN;
```

Reply:
I think we should define a system constant for the 800. Might as well do it now.

Reply:
I think we should define a system constant for the 800. Might as well do it now.

Reply:
I prefer your approach @denravonska. I am putting the constant in miner.h as

// Note the below constant controls the minimum value allowed for post
// split UTXO size. It is int64_t but in GRC so that it matches the entry in the config file.
// It will be converted to Halfords in GetNumberOfStakeOutputs by multiplying by COIN.
static const int64_t MIN_STAKE_SPLIT_VALUE_GRC = 800;

I have to do it this way to avoid a template error, because the GetArg is int64_t.